### PR TITLE
LPS-150975 Filter by lastLoginDate in getAccountUserAccountsByExternalReferenceCode

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-150975](https://issues.liferay.com/browse/LPS-150975).<br/><br/>/cc @pei-jung @erickmonteiroo @ChrisKian @patriciajeanperez @epark18 @david-truong

### CODE REVIEW

**Comment by drewbrokke** [comment]
@joseabelenda @victortrajano please have a look at these changes to see if they meet your needs. Thank you!

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by victortrajano** [comment]
> @joseabelenda @victortrajano please have a look at these changes to see if they meet your needs. Thank you!

Hey @drewbrokke, I appreciate your time, we have a specific use case where we need to filter by lastLoginDate eq null, I tested with your code, but in this case, it doesn't work. Currently, we use GraphQL to make the requests.

**Comment by drewbrokke** [comment]
@victortrajano would it work to use the [second filter in the testcase](https://github.com/liferay-usm/liferay-portal/pull/512/commits/e0b4b7920de1b02aa0ff4b2850156f8b35ddc626#diff-fa9a7684313c42cad2a94822bb7069d13bb3fcf84e05a4ebf4045bd7c8abc55fR345=)? I tried to send null, but I don't think the date-time field supports passing null as a value.

**Comment by drewbrokke** [comment]
ci:reevaluate:1267401_2537

**Comment by victortrajano** [comment]
> 



> @victortrajano would it work to use the [second filter in the testcase](https://github.com/liferay-usm/liferay-portal/pull/512/commits/e0b4b7920de1b02aa0ff4b2850156f8b35ddc626#diff-fa9a7684313c42cad2a94822bb7069d13bb3fcf84e05a4ebf4045bd7c8abc55fR345=)? I tried to send null, but I don't think the date-time field supports passing null as a value.

Yep, this works for us, I believe that we can use it this way too. Thank you!

**Comment by epark18** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
@victortrajano I just pushed a change that should enable using `null` with a date field. 

@cgoncas FYI please see the last few commits with changes to `portal-odata-impl`. LMK if it needs to be someone else looking at it.

**Comment by drewbrokke** [comment]
ci:stop

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

File: modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImplTest.java [hunk 1]
```diff
@@ -381,6 +413,23 @@ public void testVisitBinaryExpressionOperationWithNotEqualOperationAndNullValue(
 		Assert.assertEquals("*", queryTerm.getValue());
 	}
 
+	@Test
+	public void testVisitBinaryExpressionOperationWithNotEqualOperationAndNullValueForDateField() {
+		Map<String, EntityField> entityFieldsMap =
+			_entityModel.getEntityFieldsMap();
+
+		for (String fieldName : new String[] {"date", "dateTime"}) {
```
Comment (cgoncas) [inline modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImplTest.java:421]: Hey @drewbrokke !!

What do you think about create two separate test for date and dateTime?

So in the future, if this test fails, we can have a easy way to know what have happened?

Thanks!!

Comment (cgoncas) [reply modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImplTest.java]: Beautifully done

**Comment by drewbrokke** [comment]
See https://github.com/liferay-usm/liferay-portal/pull/518

**Pull request closed without merging**